### DISCUSSION TITLE
missing attribute when using join() + one()

### DISCUSSION THREAD

**Discussion by dels07** [yiisoft/yii2#19550] [Q&A] [answered]
Hi there,

I don't know if this is correct behavior of Yii Active Record, consider I have this code

```php
$post = Post::find()
    ->alias('p')
    ->select(['p.*', 'COUNT(c.id) AS comment_count'])
    ->join('LEFT JOIN', 'comments c', 'p.id = c.post_id')
    ->groupBy('c.id')
    ->one();
``` 

I cannot access `$post->comment_count`, but when I use `->asArray()->one`, I can access `$post['comment_count']`, is it possible to return as Post model while having access to `comment_count`? As this can be used for validation, example

``` php
// $post from code above
if ($post->comment_count != 0) {
    throw new UnprocessableEntityHttpException('Cannot delete post with comment(s)');
}

return $post->delete();
```

**Comment by bizley** [comment, marked as answer]
I guess you would have to add that `comment_count` as a property first.

**Comment by WinterSilence** [comment]
[read manual](https://www.yiiframework.com/doc/guide/2.0/en/db-active-record#selecting-extra-fields)
